### ISSUE TITLE
Don't set id="react-select-2-input" on all inputs

### ISSUE THREAD

**Issue by aidanlister** [JedWatson/react-select#4328]
Should we leave the id attribute off? Otherwise you'll get DOM errors:

<img width="775" alt="Screen Shot 2020-12-13 at 10 09 09 am" src="https://user-images.githubusercontent.com/54896/101998904-8ccc0100-3d2b-11eb-9593-06762c6788a8.png">

**Comment by Rall3n** [comment]
@aidanlister It seems like you have a problem with the `id` creation for the search input.

The `id`s should count up with every new instance of the `Select` component by default (`react-select-{n}-input`). You having two input elements with the same `id` means there is some kind of error.

Unfortunately I can´t recreate this problem. Would you mind sharing a [minimal reproducible example using CodeSandbox](https://codesandbox.io/s/react-select-v3-sandbox-jgdch)?

**Comment by KarimDaghari** [comment]
I used the `instanceId` prop like so: `instanceId={normalizedId}`. It got rid the warning(s) for me. 

I think the way it works is that react-select tries to keep an internal count of all its instances (hence the 2) but sometimes, for some (unknown to me) reason it doesn't work as expected.

**Comment by Methuselah96** [comment]
I've also intermittently experienced this without any code changes.

**Comment by aidanlister** [comment]
@Methuselah96 yes, it's consistently an issue. I won't have time to make a minimal example but we get this on every page where we have multiple select inputs.

We get it both on pages where we have a single react root, and on pages we don't.

Do we need that id set at all? Is there a way to disable it?

**Comment by ebonow** [comment]
@aidanlister we can investigate more to better understand why this is happening.  I personally have not come across this. 

As a work-around, I would offer you try something like lodash/uniqueId and pass that in as the `instanceId` prop to get around these warnings until we can diagnose why this is occurring and come up with a fix for it.

If you can provide a working example, when you have the opportunity, it will help us better identify the root cause. Thanks!

**Comment by aidanlister** [comment]
I’m curious why it is a thing- what is that unique ID used for? Would it be better to have id= generation disabled by default or disabled via a flag?

**Comment by ebonow** [comment]
@aidanlister Sorry but I still have not been able to reproduce this error such that the instance id repeats.

The instanceId is used for generating unique id's for the input (as well as other components within it) when one is not provided to the Select

**Comment by aidanlister** [comment]
@ebonow it seems easy to replicate, just initialise the widget twice in different react roots. My question is why do you need an ID at all? It's not a required attribute, what's the use case?

**Comment by ebonow** [comment]
@aidanlister, I looked over the code for a bit and it's not immediately clear to me. There is an open PR to support WIA-ARIA which would require unique id's in order to establish some relationships (ie: aria-owns). It's not clear why it's required but I'll keep it in mind as I keep digging through code.

**Comment by mellis481** [comment]
> it seems easy to replicate, just initialise the widget twice in different react roots.

@aidanlister Bingo. I'm frankly shocked to learn that this is how react-select was implemented.  It creates [WCAG 4.1.1](https://www.w3.org/WAI/WCAG21/Understanding/parsing.html) accessibility violations for every microfrontend-architected application.

CC: @dgreene1

**Comment by dgreene1** [comment]
@mellis481 If you override instance ID with a GUID perhaps it’s something that each MicroFrontend can handle via a wrapper in the component library. But we can talk about this more next week when I’m back from PTO. At this point I don’t know enough about the cause or the symptoms. So can you share the a11y errors here for us all?

**Comment by cda210** [comment]
I played around with this a little bit -  for those needing a viable solution with existing props - here is a snippet of what worked for me:

`export default () => (
  <>
    <label name="inputId">Label</label>
    <Select
      components={{ MultiValueContainer }}
      id={uuid}
      inputId={`inputId`}
      instanceId={uuid}
      defaultValue={[colourOptions[4], colourOptions[5]]}
      isMulti
      options={colourOptions}
    />
  </>
);`

CC: @dgreene1 @mellis481
